### PR TITLE
[WIP]started working on logic analyzer

### LINKED ISSUE #1825: Logic Analyser Not working as expected
**Actual Behaviour**
Currently logic analyser is giving weird outputs

**Expected Behaviour**
Logic analyser should give output similar to python library and conventional logic analyser

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1825

**Changes**: WIP

**Screenshot/s for the changes**:  WIP

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [ ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [ ] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [ ] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**: WIP

### CODE REVIEW

File: app/src/main/java/io/pslab/communication/digitalChannel/DigitalChannel.java [hunk 1]
```diff
@@ -67,18 +67,21 @@ public void loadData(LinkedHashMap<String, Integer> initialStates, double[] time
...
+                factor = 1;
+                break;
+            case 3:
                 factor = 4;
                 break;
             default:
                 factor = 1;
         }
-        for (int i = 0; i < this.timestamps.length; i++) this.timestamps[i] /= factor;
+        for (int i = 0; i < this.timestamps.length; i++) this.timestamps[i] *= factor;
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/communication/digitalChannel/DigitalChannel.java]: Kinda same thing. Multiply and Divide is same as Divide and Multiply. Just FYI

Comment (neel1998) [reply app/src/main/java/io/pslab/communication/digitalChannel/DigitalChannel.java]: No but when prescalar is 3, we were before still divinding by 4, which according to Python library we are suppose to multiply . That's why I made this change.

**Comment by CloudyPadmal** [comment]
@neel1998 any update?

**Comment by neel1998** [comment]
I have checked the code with Python and have made necessary changes...The issue is coming in the USB communication handler. I have talked with @harsh-2711 and @viveksb007 as well regarding this. Apparently LA was working fine sometimes back but some changes made into science lab or other class is causing the issues. I am trying to get LA back to the original state as soon as possible. Apologies for the delay. Will try my best to get this done.

**Comment by CloudyPadmal** [comment]
Nice. Like I said in the evaluation feedback; this and I2C they were all functional. But broken as we progressed. The reason is we don't have a testing mechanism to run a full test after each PR. PRs being 100+ lines of code, it's hard to catch a break in another by manual testing.

As a workaround, make sure to test the whole app by the developer himself even when making a small change and then make the PR.

**Comment by neel1998** [comment]
Local branch got messy... Closing this one

**Pull request closed without merging**